### PR TITLE
Retraining pipeline for classification

### PR DESCRIPTION
**Pull request by MercyPrasanna**

### CODE REVIEW

File: Pipelines/ModelRetraining/register_model_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+from azureml.core import Workspace
+from azureml.train.automl.run import AutoMLRun
+from azureml.pipeline.core import PipelineRun
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--ds_name", help="input dataset name")
+parser.add_argument("--model_name", help="model name to register")
+parser.add_argument("--model_path",  help="model file from training run")
+parser.add_argument("--metrics_path",  help="metrics file from training run")
+parser.add_argument('--register_always', default=False, type=lambda x: (str(x).lower() == 'true'))
```
Comment (martinezgjuan) [inline Pipelines/ModelRetraining/register_model_pipeline.py:16]: Minor: we use the following in other bool argument parsing:
`type=lambda x: bool(strtobool(str(x)))`

File: Pipelines/ModelRetraining/register_model_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+run = Run.get_context()
+ws = None
+if type(run) == _OfflineRun:
+    ws = Workspace.from_config()
+else:
+    ws = run.experiment.workspace
+
+if args.model_name == 'automl_image_model':
+    print('No non default model name provided from the arguments. Using auto-generated model name from automl child run.')
```
Comment (martinezgjuan) [inline Pipelines/ModelRetraining/register_model_pipeline.py:33]: Isn't just saying "Default model name provided" simpler?

File: Pipelines/ModelRetraining/AutoMLImage_MultiClass_RetrainingPipeline.ipynb [hunk 3]
```diff
@@ -0,0 +1,1099 @@
...
+    "### Note: Only Python 3.6 and 3.7 are supported for this feature."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install --upgrade  \"azureml-pipeline-steps\" \"azureml-train-core<0.1.1\" \"azureml-train-automl<0.1.1\" \"azureml-contrib-dataset<0.1.1\" --extra-index-url \"https://azuremlsdktestpypi.azureedge.net/automl_for_images_private_preview/\""
```
Comment (martinezgjuan) [inline Pipelines/ModelRetraining/AutoMLImage_MultiClass_RetrainingPipeline.ipynb:38]: Why is this not also <0.1.1 ?

File: Pipelines/ModelRetraining/AutoMLImage_MultiClass_RetrainingPipeline.ipynb [hunk 4]
```diff
@@ -0,0 +1,1099 @@
...
...1LdW3a2Ouik4lHVIjWDojT4i1JlFCtpLSHNE+4jY8bG/uOrY7lsejE+oCyn03N6A/hQP/2elDu8opySNcRYrq2FyealsOfUPOS/TCkrcG9paHJKQn2qJJI5ADrxDt1TU5qUKWY3+nyx732Hq6xNIQ0J+lvrhowHCrjt5Igybp1UCZZJUufHTKRxtlhorcWPP6RQFAKcBCdhWgrfHpNz6by71SLEJen2UdS0tWaXVFx+RwHJW3ElThQClJ8AgD2geD064VBOcyUPXTbCC+hLch6fFLMORE22tkAg8m3gvaSPjmoeFJPjF3EgPdpHItvi+CTLqI9OckMzZktKmmHmuW2/OwoKStQPuBJ0NEjzwTla0P4TCbz74647RjKqg/x6xdo90nl64wWeSZWEqyjtyFsvs164shxxpLogpGk8XkcfUQpQWFoUCvYJ8bB68fh1zaiyVyZ2s7tOtxJUNHByxb4od5oJ4rPgKcSQTyWNkHivXk7YcrxDG+41VWdz+21lYQjYwlh4All9t+N+5LxPhz00KSnfIk+07UTorGZdn4eQ4TWPUG5OQ/SlEa0S+GFoSDtxwoT7vTSVemlRI0PalPz1AtVlYwXXVaRcg7EGdsTTUZZl1SGGEayYBAEgiNz8/fjxd1eE4hmrWFZG/wDSsxHyrH8hqEhTNa+sc20OhXhSeRHkaO/ck/HVhy/xMZnIqZGN90sNgoubVK1puKh8Fix2hLbjEhPu0eKQoqV7ebY5EEk9L1tbV2NUUDGcswebOfjyobBtxXJelPLU0S4+tt4pQoghCN+BoJ5L0fHV/NtcryF+S9BjwHfQUxA9dDfpWZClIUFpSB6DhPg7AIJCdnQPQHaagSgKeUSq32MNU9fmzjpTTICUX348p6/2x7w/te/lOYxXYFkh2HKalGXjCHm2kIkAtlDjfuT6zCkaWlAJKFIUN8TslanEqTshlMu5alVjlpICYzAgzYbkqCz+4vlLSlhK3NgJAJKQCpWiUjpMyGuYyWmTk6npKLWvDMZUG00tuNxSOUYp4Ap8q2FJ0QnZ+xUHyTlfZg4Ezd0nZ9Dsplh5q/q+TbqGnUpQskhxIUpBUCNEg+CUqBT5OMwDaginZ1CfXb74YUVlC3EKXW1OhWmN7RPL9f0J/kGa3cX0KpDyK+7LyGZ0d5Mp1ts+5SUEjmgnfMgJUSAop5gaHXaOHjMev
```
Comment (martinezgjuan) [inline Pipelines/ModelRetraining/AutoMLImage_MultiClass_RetrainingPipeline.ipynb:388]: Align arguments

Comment (martinezgjuan) [inline Pipelines/ModelRetraining/AutoMLImage_MultiClass_RetrainingPipeline.ipynb:426]: Align arguments

Comment (martinezgjuan) [inline Pipelines/ModelRetraining/AutoMLImage_MultiClass_RetrainingPipeline.ipynb:430]: Align arguments

Comment (martinezgjuan) [inline Pipelines/ModelRetraining/AutoMLImage_MultiClass_RetrainingPipeline.ipynb:548]: Indent arguments

Comment (martinezgjuan) [inline Pipelines/ModelRetraining/AutoMLImage_MultiClass_RetrainingPipeline.ipynb:823]: Align arguments

Comment (martinezgjuan) [inline Pipelines/ModelRetraining/AutoMLImage_MultiClass_RetrainingPipeline.ipynb:825]: Align arguments

File: Pipelines/ModelRetraining/register_model_pipeline.py (hunk 2 above)
Comment (MercyPrasanna) [reply Pipelines/ModelRetraining/register_model_pipeline.py:33]: Yes agreed. Updated the same.

File: Pipelines/ModelRetraining/AutoMLImage_MultiClass_RetrainingPipeline.ipynb (hunk 3 above)
Comment (MercyPrasanna) [reply Pipelines/ModelRetraining/AutoMLImage_MultiClass_RetrainingPipeline.ipynb:38]: yes added the version check

File: Pipelines/ModelRetraining/register_model_pipeline.py (hunk 1 above)
Comment (MercyPrasanna) [reply Pipelines/ModelRetraining/register_model_pipeline.py:16]: Good to know of this option we have. Leaving this for now as limited choices of true/false should be fine.

**Review by raduk — APPROVED** [review]

**Comment by swatig007** [comment]
lgtm

File: Pipelines/ModelRetraining/register_model_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+# Get the metrics data
+try:
+    with open(args.metrics_path) as f:
+        metrics_data = json.load(f)
+except Exception as e:
+    print("Could not load the metrics file.")
+    raise
+
+# Retrieve the hd run's data
+for key in metrics_data.keys():
```
[the remark this answers is not in the document]
Comment (MercyPrasanna) [reply Pipelines/ModelRetraining/register_model_pipeline.py]: Yes correct. Made the changes.

**Pull request merged**